src/features/accounts/WalletList.tsx

```diff
@@ -0,0 +1,378 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import Text from '@components/Text'
+import { useTranslation } from 'react-i18next'
+import { ReAnimatedBox } from '@components/AnimatedBox'
+import { FadeIn } from 'react-native-reanimated'
+import Box from '@components/Box'
+import { Image, SectionList } from 'react-native'
+import { useAccountStorage } from '@config/storage/AccountStorageProvider'
+import { NetTypes } from '@helium/address'
+import { CSAccount } from '@config/storage/cloudStorage'
+import TouchableContainer from '@components/TouchableContainer'
+import AccountIcon from '@components/AccountIcon'
+import { ellipsizeAddress } from '@utils/accountUtils'
+import { useColors, useSpacing } from '@config/theme/themeHooks'
+import SmallAdd from '@assets/svgs/smallAdd.svg'
+import Checkmark from '@assets/svgs/checkmark.svg'
+import { useNavigation } from '@react-navigation/native'
+import {
+  HELIUM_DERIVATION,
+  keypairFromSeed,
+  solanaDerivation,
+} from '@hooks/useDerivationAccounts'
+import { getSecureAccount } from '@config/storage/secureStorage'
+import * as bip39 from 'bip39'
+import { useOnboarding } from '@features/onboarding/OnboardingProvider'
+import Toast from 'react-native-simple-toast'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { ServiceSheetNavigationProp } from 'src/app/services/serviceSheetTypes'
+import CircleLoader from '@components/CircleLoader'
+import ScrollBox from '@components/ScrollBox'
+import {
+  OnboardingSheetRef,
+  OnboardingSheetWrapper,
+} from '@features/onboarding/OnboardingSheet'
+import { NavBarHeight } from '@components/ServiceNavBar'
+import { AccountsServiceNavigationProp } from './accountServiceTypes'
+
+const WalletList = () => {
+  const { t } = useTranslation()
+  const [switchingAccounts, setSwitchingAccounts] = useState<
+    CSAccount | undefined
+  >()
+  const spacing = useSpacing()
+  const { setOnboardingData, onboardingData } = useOnboarding()
+  const colors = useColors()
+  const navigation = useNavigation<
+    AccountsServiceNavigationProp & ServiceSheetNavigationProp
+  >()
+  const { sortedAccounts, currentAccount, setCurrentAccount, accounts } =
+    useAccountStorage()
+  const { bottom } = useSafeAreaInsets()
+  const onboardingSheetRef = useRef<OnboardingSheetRef>(null)
+
+  const handleAddSub = useCallback(
+    async (acc: CSAccount) => {
+      onboardingSheetRef?.current?.show('sub-wallet')
+
+      // wait 1 second
+      await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
y tho?

Reply:
bip39.mnemonicToSeedSync takes up the UI thread so we need to show the loading state before it takes up the UI thread